packages/core/src/components/widgets/CheckboxesWidget.tsx

```diff
@@ -0,0 +1,68 @@
+import { WidgetProps } from "@rjsf/utils";
+import React from "react";
+
+function selectValue(value: any, selected: any[], all: any[]) {
+  const at = all.indexOf(value);
+  const updated = selected.slice(0, at).concat(value, selected.slice(at));
+  // As inserting values at predefined index positions doesn't work with empty
+  // arrays, we need to reorder the updated selection to match the initial order
+  return updated.sort((a, b) => Number(all.indexOf(a) > all.indexOf(b)));
+}
+
+function deselectValue(value: any, selected: any[]) {
+  return selected.filter((v) => v !== value);
+}
+
+function CheckboxesWidget<T = any, F = any>({
+  id,
+  disabled,
+  options: { inline = false, enumOptions, enumDisabled },
+  value,
+  autofocus = false,
+  readonly,
+  onChange,
+}: WidgetProps<T, F>) {
+  return (
+    <div className="checkboxes" id={id}>
+      {Array.isArray(enumOptions) &&
+        enumOptions.map((option, index) => {
+          const checked = value.indexOf(option.value) !== -1;
+          const itemDisabled =
+            enumDisabled && enumDisabled.indexOf(option.value) != -1;
+          const disabledCls =
+            disabled || itemDisabled || readonly ? "disabled" : "";
+          const checkbox = (
+            <span>
+              <input
+                type="checkbox"
+                id={`${id}_${index}`}
+                checked={checked}
+                disabled={disabled || itemDisabled || readonly}
+                autoFocus={autofocus && index === 0}
+                onChange={(event) => {
+                  const all = enumOptions.map(({ value }) => value);
+                  if (event.target.checked) {
+                    onChange(selectValue(option.value, value, all));
+                  } else {
+                    onChange(deselectValue(option.value, value));
+                  }
+                }}
```

Comment:
Might I suggest refactoring this out into a fat-arrow function similar to what you see in the `BaseInputTemplate` while adding the proper type for the `event`?